dashboard/app/models/stage.rb

```diff
@@ -75,6 +75,32 @@ def summarize
         levels: script.script_levels.to_a.select{|sl| sl.stage_id == id}.map(&:summarize),
     }
 
+    # Go through all levels.  If we find an assessment with a per_page value,
+    # then it's a long assessment: we assume it's the final level in the
+    # stage, and generate some placeholder levels for the subsequent pages
+    # of the long assessment.  Each of those levels will have the same basic
+    # URL as the first, but with ?page=1, ?page=2, etc.
+
+    levels = stage_data[:levels]
+
+    # We might append additional levels, so iterate over the original set.
+    levels[0..levels.length].each do |level|
+      if level[:kind] == "assessment"
```

Comment:
I think it may be cleaner to refactor this so that it's an instance method on LevelGroup so you can deal with the nice objects instead of a hash that you need to interpret again. I think this will also allow you do reuse more code between this and the _level_group.haml file below. However I am guessing that if you end up having to change this to do level-builder-specified groups instead of pages I think it will end up being complicated enough that you will be forced to use the model anyway :)